storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -135,7 +137,28 @@ public static int samplingRate(Map conf) {
         throw new IllegalArgumentException("Illegal topology.stats.sample.rate in conf: " + rate);
     }
 
-    // public static mkStatsSampler // depends on Utils.evenSampler() TODO, this is sth we need to do after util
+    public static Callable<Boolean> evenSampler(final int samplingFreq) {
+        final int start = 0;
+        final Random random = new Random();
+        final MutableInt curr = new MutableInt(-1);
+        final MutableInt target = new MutableInt(random.nextInt(samplingFreq));
```

Comment:
will do. Out of Curiosity, are the calls to emitSampler object done in single thread?